### PR TITLE
Add stack query method

### PR DESCRIPTION
**Pull request by jpslav**
See README

### NOTE

This PR's query method returns `Stack` objects.  This is the first use of a kind of read-only stack object -- all other uses so far had been of stacks that were to be created, updated, etc.  As such, not everything in the `Stack` object really applies and works.  I think we should do some work to split the `Stack` object into two: a `SimpleStack` and a `DeployableStack`, the latter of which has all the functionality and is the thing that is configured from a deployment.  This is written up in #58.

### TODO

* [x] Specs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -345,6 +345,18 @@ instance) and then getting its desired capacity.
 
 No options here besides `wait`.
 
+#### `query`
+
+A class method `query` is provided on the `Stack` class to find stacks matching certain criteria:
+
+* `regex`: only queries whose name matches this regex will be returned
+* `regions`: only stacks in these regions will be queried (defaults to US regions)
+* `active`: if true, excludes stacks that have been deleted or that failed to create
```
Comment (PiotrKozlowski) [inline README.md:354]: I think we should add default values to the documentation

File: lib/openstax/aws/stack.rb [hunk 2]
```diff
@@ -361,10 +361,52 @@ def exists?
...
+
+    def active_statuses
+      all_statuses - %w(CREATE_FAILED DELETE_COMPLETE)
+    end
+
     def defines_secrets?
       !secrets_blocks.empty?
     end
 
+    def self.query(regex: /.*/, regions: %w(us-east-1 us-east-2 us-west-1 us-west-2), active: true)
```
Comment (PiotrKozlowski) [inline lib/openstax/aws/stack.rb:399]: You are not using `regex` in this method

Comment (jpslav) [reply lib/openstax/aws/stack.rb:399]: Lol

Comment (jpslav) [reply lib/openstax/aws/stack.rb]: Fixed this.

File: README.md (hunk 1 above)
Comment (jpslav) [reply README.md]: Done.

**Pull request merged**